File: android_storj_uploader/app/src/main/java/com/example/storjapp/config/UploadConfig.kt
package com.example.storjapp.config

/**
 * Configuration constants for upload limits and batch processing
 */
object UploadConfig {
    /**
     * Maximum number of images allowed in upload queue
     */
    const val MAX_IMAGE_UPLOAD_LIMIT = 1000

    /**
     * Maximum number of videos allowed in upload queue
     */
    const val MAX_VIDEO_UPLOAD_LIMIT = 100

    /**
     * Batch size for parallel uploads
     */
    const val UPLOAD_BATCH_SIZE = 10

    /**
     * Time window in hours for recent media detection
     */
    const val RECENT_MEDIA_HOURS = 24
}

File: storj_uploader_frontend_container_app/src/config/uploadConfig.ts
/**
 * Upload configuration constants
 */
export const UPLOAD_CONFIG = {
  /**
   * Maximum number of images allowed in upload queue
   */
  MAX_IMAGE_UPLOAD_LIMIT: 1000,

  /**
   * Maximum number of videos allowed in upload queue
   */
  MAX_VIDEO_UPLOAD_LIMIT: 100,

  /**
   * Maximum files per single upload operation
   */
  MAX_FILES_PER_UPLOAD: 10,

  /**
   * Warning threshold (percentage of limit)
   */
  WARNING_THRESHOLD: 0.8, // 80%
} as const;

export type UploadConfig = typeof UPLOAD_CONFIG;
